js/ssai_pulse.js

```diff
@@ -237,19 +273,20 @@ OO.Ads.manager(function(_, $)
       {
         adMode = true;
         this.currentAd = ad;
-        if (this.currentAd.ad)
-        {
-          this.currentAd.ad.id3AdId = this.currentId3Object.adId;
+        if (ad.ad && ad.ad.data) {
+          this.adIdDictionary[ad.ad.data.id].curAdId = ad.id;
```

Comment:
@cacel @ipadilla4 is `ad.ad.data.id` guaranteed to exist here? I suspect it should, (especially if `ad.ad.data` exists) but we might want to add a null check just in case

Reply:
@dallen42 it should exist but I agree to verify just in case, I will add that change.